FLT/HaarMeasure/ConcreteCharCalculations.lean

```diff
@@ -0,0 +1,79 @@
+import Mathlib.Analysis.Complex.ReImTopology
+import Mathlib.MeasureTheory.Measure.Lebesgue.EqHaar
+import Mathlib.RingTheory.Complex
+import Mathlib.RingTheory.Localization.NumDen
+import Mathlib.Analysis.Normed.Group.Ultra
+import FLT.HaarMeasure.DistribHaarChar
+import FLT.Mathlib.Analysis.Complex.Basic
+import FLT.Mathlib.Data.Complex.Basic
+import FLT.Mathlib.LinearAlgebra.Determinant
+import FLT.Mathlib.MeasureTheory.Group.Action
+import FLT.Mathlib.NumberTheory.Padics.PadicIntegers
+import FLT.Mathlib.NumberTheory.Padics.MeasurableSpacePadics
+import FLT.Mathlib.RingTheory.Norm.Defs
+
+open Real Complex Padic MeasureTheory Measure NNReal Set TopologicalSpace
+open scoped Pointwise ENNReal Localization
+
+lemma distribHaarChar_complex (z : ℂˣ) : distribHaarChar ℂ z = ‖(z : ℂ)‖₊ ^ (-2 : ℤ) := by
+  refine distribHaarChar_eq_of_measure_smul_eq_mul (s := Icc 0 1 ×ℂ Icc 0 1) (μ := volume)
+    (measure_pos_of_nonempty_interior _ <| by simp [interior_reProdIm]).ne'
+    (isCompact_Icc.reProdIm isCompact_Icc).measure_ne_top ?_
+  have key : ((LinearMap.mul ℂ ℂ z).restrictScalars ℝ).det = ‖z.val‖₊ ^ 2 := by
+    refine Complex.ofReal_injective ?_
+    rw [LinearMap.det_restrictScalars]
+    simp [Algebra.norm_complex_apply, normSq_eq_norm_sq, zpow_ofNat]
+  have := addHaar_preimage_linearMap (E := ℂ) volume
+    (f := (LinearMap.mul ℂ ℂ z).restrictScalars ℝ)
+  simp [key] at this
+  convert this _ using 2
+  · symm
+    exact Set.preimage_smul z _
+  · simp [← Real.ennnorm_eq_ofReal_abs, ← Complex.norm_eq_abs, zpow_ofNat]
+
+lemma distribHaarChar_real (x : ℝˣ) : distribHaarChar ℝ x = ‖(x : ℝ)‖₊⁻¹ := by
```

Comment:
The whole point is that this `distribHaarChar` is a way of recovering the norm, not a way of recovering the inverse of the norm.

Reply:
I see a few options then:
* Use `(distribHaarChar G x)⁻¹`. Likely painful
* Keep assuming that `G` acts on `A` but make `distribHaarChar G` a group hom with domain `Gᵐᵒᵖ`. Likely painful
* Keep `distribHaarChar G` a group hom with domain `G` but assume that `Gᵐᵒᵖ` acts on `A` instead of `G`. Probably the best solution?

Anyway, I think this isn't a discussion for the current PR. The definition is already on master, this PR only computes it, and the refactor to invert everything should be its own PR and can be done after the current PR.

Reply:
I forced it to be an issue in this PR by fixing the definition in #274 (now merged).